### PR TITLE
Adding Time Selector

### PR DESCRIPTION
**Pull request by stevenernst131**
image

image

### CODE REVIEW

File: AngularApp/src/app/diagnostic-data/components/detector-view/detector-view.component.ts [hunk 1]
```diff
@@ -45,11 +46,30 @@ export class DetectorViewComponent implements OnInit {
...
+  constructor(@Inject(DIAGNOSTIC_DATA_CONFIG) config: DiagnosticDataConfig, private telemetryService: TelemetryService, private detectorControlService: DetectorControlService) {
     this.isPublic = config && config.isPublic;
   }
 
   ngOnInit() {
+    this.loadDetector();
+
+    this.detectorControlService.update.subscribe(validUpdate => {
+      if (validUpdate) {
+
```
Comment (cindylovescoding) [inline AngularApp/src/app/diagnostic-data/components/detector-view/detector-view.component.ts:58]: Is there any operations should be included here? What will happhen if the update is not valid?

File: AngularApp/src/app/diagnostic-data/services/detector-control.service.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+  constructor(@Inject(DIAGNOSTIC_DATA_CONFIG) config: DiagnosticDataConfig) {
+    this.internalClient = !config.isPublic;
+  }
+
+  public get update() {
+    return this._refresh;
+  }
+
+  public setDefault() {
```
Comment (cindylovescoding) [inline AngularApp/src/app/diagnostic-data/services/detector-control.service.ts:61]: When will this be called to set default duration?

File: AngularApp/src/app/diagnostic-data/components/detector-control/detector-control.component.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  setManualDate() {
+    this.detectorControlService.setCustomStartEnd(this.startTime, this.endTime)
+  }
+}
+
+@Pipe({
+  name:'internal',
+  pure: false
+})
+export class InternalPipe implements PipeTransform {
```
Comment (cindylovescoding) [inline AngularApp/src/app/diagnostic-data/components/detector-control/detector-control.component.ts:51]: Could you help explain when will this InternalPipe be used:) Why do we specify "3d" duration as internalOnly?

**Comment by cindylovescoding** [comment]
Time-select looks really good!!

File: AngularApp/src/app/diagnostic-data/components/detector-view/detector-view.component.ts (hunk 1 above)
Comment (stevenernst131) [reply AngularApp/src/app/diagnostic-data/components/detector-view/detector-view.component.ts:58]: I am not really using the valid update yet. That can be future improvement. If they set a time that is too long for example, I don't want to make a call to the back end.

File: AngularApp/src/app/diagnostic-data/services/detector-control.service.ts (hunk 2 above)
Comment (stevenernst131) [reply AngularApp/src/app/diagnostic-data/services/detector-control.service.ts:61]: This is called when no time is specified. I will be using it in App Service Diagnostics specifically.

File: AngularApp/src/app/diagnostic-data/components/detector-control/detector-control.component.ts (hunk 3 above)
Comment (stevenernst131) [reply AngularApp/src/app/diagnostic-data/components/detector-control/detector-control.component.ts:51]: Our detectors are not built ready to handle 3 days as input. I am enabling it internally to see how it goes as a trial period.

Comment (stevenernst131) [reply AngularApp/src/app/diagnostic-data/components/detector-control/detector-control.component.ts:51]: The pipe works by taking the list as input and filtering it appropriately

**Review by cindylovescoding — APPROVED** [review]

**Review by ShekharGupta1988 — APPROVED** [review]

File: AngularApp/src/app/diagnostic-data/components/detector-control/detector-control.component.html [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      <div class="control date-control">
+        <input class="date-input" [(ngModel)]="endTime" (keyup.enter)="setManualDate()" />
+      </div>
+      <div class="control control-static">
+        UTC
+      </div>
+    </div>
+
+    <div class="control-group" *ngIf="isInternal">
+        <div class="control" (click)="detectorControlService.toggleInternalExternal()">
```
Comment (puneetg1983) [inline AngularApp/src/app/diagnostic-data/components/detector-control/detector-control.component.html:44]: >toggleInternalExternal [](start = 61, length = 22)

I am assuming the purpose of this method is to be able to see in AppLens what we show externally ?

File: ApplensBackend/Services/DiagnosticRoleClientService/DiagnosticRoleClient.cs [hunk 5]
```diff
@@ -66,30 +66,28 @@ private HttpClient InitializeClient()
                 MaxResponseContentBufferSize = Int32.MaxValue
             };
 
-            client.DefaultRequestHeaders.Add("internal-applens", "true");
+            client.DefaultRequestHeaders.Add("x-ms-internal-client", "true");
```
Comment (puneetg1983) [inline ApplensBackend/Services/DiagnosticRoleClientService/DiagnosticRoleClient.cs:69]: >x-ms-internal-client [](start = 46, length = 20)

Any specific reason we changed this from internal-applens? Is this some kind of convention that we are trying to follow ?

**Review by puneetg1983 — APPROVED** [review]
:shipit:

**Pull request merged**